tests/lib/managers/folders-test.js

```diff
@@ -276,4 +276,20 @@ describe('Folders', function() {
 		});
 	});
 
+	describe('deletePermanently()', function() {
+
+		it('should call BoxClient defaultResponseHandler method with the callback when response is returned', function(done) {
+			sandbox.mock(boxClientFake).expects('defaultResponseHandler').returns(done);
```

Comment:
I would change this to either `.withArgs(done).returns(done)` or simply `.returnsArg(0)` to capture the expected behavior around `defaultResponseHandler()`, namely that is is passed a callback and produces a new callback that should be called once the operation is complete.